### DISCUSSION TITLE
Global Filters in MkDocs Material Theme

### DISCUSSION THREAD

**Discussion by sn0wcat** [squidfunk/mkdocs-material#6630] [General]
Is it possible to have  "global filters" (based on configuration, tag or similar) which would hide parts of the documentation site based on the selected role of the user? 

We want to have a behaviour where the user selects a "view" on the starting page and this filters out in the navigation elements everything which is not relevant to that "view". 

(The reason for this is that we have a large documentation site and we want to provide specific, dedicated views on the documentation dependent on the user selection).

**Comment by alexvoss** [comment]
I have not come across anything like this so far but perhaps others know of something. 

Assuming that there is no ready-made solution, can you elaborate a bit more on how you would imaging this working? Are we talking about different versions of the site that get built at compile-time or dynamic selection? Both, I think, would look very different both in terms of implementation and user experience.

**Reply by sn0wcat** [reply]
@alexvoss i was thinking about a dynamic selection where a documentation reader can dynamically pick a view (or switch it off). We are not aiming for compile time filtering (which could be implemented easily with different mkdocs files).

**Reply by alexvoss** [reply]
This does strike me as a potentially useful feature. With Material, we have a thin but constant stream of people who get confused about what or is not available in the public version. While I am pretty sure we would not want to outright hide Insiders features, some different representation when people select that they are using the public version might make sense (highlighting use cases rather than explaining usage). If this would not work for Material then I can imagine other places where people might want to make that kind of decision. 

I imagine this can be implemented as a customization. I would start with the [blocks syntax](https://facelessuser.github.io/pymdown-extensions/extensions/blocks/api/#blocks-extension-api) in the Python Markdown Extensions to generate the necessary HTML - though just using HTML `div`s would be possible as well, with the [`md_in_html` extension](https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown/#markdown-in-html). The some CSS & JavaScript would need to take it from there.
